File: gymgenie-mobile/iosApp/iosApp/Components/FeaturedWorkoutCard.swift
import SwiftUI
import Shared

struct FeaturedWorkoutCard: View {
    let plan: WorkoutPlanShortResponse
    var onStart: () -> Void = {}

    private let orange = Color(red: 0.941, green: 0.439, blue: 0.188)
    private let deepInk = Color(red: 0.161, green: 0.141, blue: 0.125)
    private let softCard = Color(red: 0.953, green: 0.949, blue: 0.937)
    private let mutedText = Color(red: 0.463, green: 0.447, blue: 0.416)

    private var isAI: Bool {
        plan.createdBy.uppercased() == "AI"
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            HStack(alignment: .top) {
                Text(plan.name)
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(deepInk)
                    .lineLimit(2)
                    .frame(maxWidth: .infinity, alignment: .leading)

                if isAI {
                    Text("+ AI")
                        .font(.system(size: 11, weight: .bold))
                        .foregroundColor(.white)
                        .padding(.horizontal, 10)
                        .padding(.vertical, 4)
                        .background(Capsule().fill(orange))
                }
            }

            HStack(spacing: 8) {
                tagChip(text: "\(plan.daysCount) дней/нед.")
                if let desc = plan.description_, !desc.trimmingCharacters(in: .whitespaces).isEmpty {
                    tagChip(text: String(desc.prefix(25)))
                }
            }

            if plan.isActive {
                HStack(spacing: 6) {
                    Circle().fill(orange).frame(width: 8, height: 8)
                    Text("АКТИВНЫЙ ПЛАН")
                        .font(.system(size: 11, weight: .bold))
                        .foregroundColor(orange)
                        .kerning(0.5)
                }
            }

            Button(action: onStart) {
                Text("Начать тренировку")
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 12)
                    .background(RoundedRectangle(cornerRadius: 12).fill(orange))
            }
            .buttonStyle(.plain)
            .padding(.top, 4)
        }
        .padding(16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(RoundedRectangle(cornerRadius: 20).fill(.white))
        .overlay(
            RoundedRectangle(cornerRadius: 20)
                .stroke(plan.isActive ? orange : Color.clear, lineWidth: 2)
        )
        .shadow(color: Color.black.opacity(0.06), radius: 6, y: 2)
    }

    private func tagChip(text: String) -> some View {
        Text(text)
            .font(.system(size: 11))
            .foregroundColor(mutedText)
            .padding(.horizontal, 10)
            .padding(.vertical, 5)
            .background(Capsule().fill(softCard))
    }
}

File: gymgenie-mobile/composeApp/src/androidMain/kotlin/com/asc/gymgenie/feature/workouts/components/FeaturedWorkoutCard.kt
package com.asc.gymgenie.feature.workouts.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.asc.gymgenie.ui.theme.AccentOrange
import com.asc.gymgenie.ui.theme.DeepInk
import com.asc.gymgenie.ui.theme.MutedText
import com.asc.gymgenie.ui.theme.SoftCard
import com.asc.gymgenie.workout.WorkoutPlanShortResponse

@Composable
fun FeaturedWorkoutCard(
    plan: WorkoutPlanShortResponse,
    onStart: () -> Unit = {},
) {
    val isAI = plan.createdBy.equals("AI", ignoreCase = true)
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .shadow(4.dp, RoundedCornerShape(20.dp))
            .clip(RoundedCornerShape(20.dp))
            .background(Color.White)
            .then(
                if (plan.isActive) Modifier.border(
                    width = 2.dp,
                    color = AccentOrange,
                    shape = RoundedCornerShape(20.dp),
                ) else Modifier
            )
            .padding(16.dp),
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                text = plan.name,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = DeepInk,
                modifier = Modifier.weight(1f),
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
            )

            if (isAI) {
                Spacer(modifier = Modifier.width(8.dp))
                Text(
                    text = "+ AI",
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    modifier = Modifier
                        .clip(RoundedCornerShape(50))
                        .background(AccentOrange)
                        .padding(horizontal = 10.dp, vertical = 4.dp),
                )
            }
        }

        Spacer(modifier = Modifier.height(8.dp))

        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            TagChip(text = "${plan.daysCount} дней/нед.")
            plan.description?.let { desc ->
                val trimmed = desc.trim()
                if (trimmed.isNotEmpty()) {
                    TagChip(text = trimmed.take(25))
                }
            }
        }

        if (plan.isActive) {
            Spacer(modifier = Modifier.height(10.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                Box(
                    modifier = Modifier
                        .size(8.dp)
                        .clip(CircleShape)
                        .background(AccentOrange),
                )
                Spacer(modifier = Modifier.width(6.dp))
                Text(
                    text = "АКТИВНЫЙ ПЛАН",
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Bold,
                    color = AccentOrange,
                )
            }
        }

        Spacer(modifier = Modifier.height(14.dp))

        Button(
            onClick = onStart,
            modifier = Modifier.fillMaxWidth(),
            colors = ButtonDefaults.buttonColors(
                containerColor = AccentOrange,
                contentColor = Color.White,
            ),
            shape = RoundedCornerShape(12.dp),
            contentPadding = PaddingValues(vertical = 12.dp),
        ) {
            Text(
                text = "Начать тренировку",
                fontSize = 14.sp,
                fontWeight = FontWeight.SemiBold,
            )
        }
    }
}

@Composable
private fun TagChip(text: String) {
    Text(
        text = text,
        fontSize = 11.sp,
        color = MutedText,
        modifier = Modifier
            .clip(RoundedCornerShape(50))
            .background(SoftCard)
            .padding(horizontal = 10.dp, vertical = 5.dp),
    )
}
